scripts/dcpdig

```diff
@@ -36,9 +40,23 @@ class DcpDig:
         entities_to_show = args.show.split(',')
 
         try:
-            finder = Finder.factory(finder_name=args.component, deployment=self.deployment)
+            # TODO: switch to OAuth to simplify the following conditional blocks
+            if args.credentials:
```

Comment:
I think there might be a way to eliminate this if statement, without needing OAuth.  If we convert `args` to a dict immediately, then delete things from it as we extract them, we can pass what remains to Finder.factory() as `**args`, e.g.:
```
        args = vars(parser.parse_args())
        self.verbose = args.pop('verbose')

        self.deployment = self._choose_deployment(args)
        print(f"Using deployment {self.deployment}")

        component = args.pop('component')
        if component.startswith('@'):
            component = component[1:]

        entities_to_show = args.pop('show').split(',')

        try:
            expression = args.pop('expression')
            finder = Finder.factory(finder_name=component, deployment=self.deployment, **args)
            entity = finder.find(expression)
```

I have done this in commit https://github.com/HumanCellAtlas/dcp-diag/commit/d58095058870fb7784c6180bf8651f4263dc67e3 on branch https://github.com/HumanCellAtlas/dcp-diag/commits/spierson/upload-int-pks.  Please feel free to cherry pick this commit.